### DISCUSSION TITLE
@babel/code-frame has problem building with Vite

### DISCUSSION THREAD

**Discussion by tiye** [babel/babel#15321] [General]
Reported previously at https://github.com/sindresorhus/parse-json/issues/35 but turned out to be problem in `@babel/code-frame` .

I got error `TypeError: Cannot read properties of undefined (reading 'close')` after running Vite build using this package. Error traced to:

<img width="915" alt="image" src="https://user-images.githubusercontent.com/449224/210224669-96cf3460-74fa-4331-8045-bf8e65735e05.png">
<img width="414" alt="image" src="https://user-images.githubusercontent.com/449224/210224746-ff36a663-7166-4e93-b883-d93f377ddd50.png">
<img width="267" alt="image" src="https://user-images.githubusercontent.com/449224/210224763-050f6e51-775d-4d8f-97fc-68000f5fec03.png">

it works fine in Vite DEV demo. I guess it's because this package is referencing very old chalk 2.x which Vite does not optimizes well.

Any plan for updating chalk to a new version?

**Comment by nicolo-ribaudo** [comment]
We plan to update Chalk in Babel 8, but I suggest reporting this bug to Vite (if they aren't aware of it yet). Even when bundling an old package, the resulting bundle should not be broken.

**Reply by tiye** [reply]
tried but I didn't collect enough information for filling all the fields in the bug report form and fell into this Discussion category.
